src/nifake/unit_tests/test_library_interpreter.py

```diff
@@ -410,6 +409,61 @@ def test_read_with_warning(self):
             assert test_error_desc in str(w[0].message)
             assert f'Warning {test_error_code} occurred.' in str(w[0].message)
 
+    def test_library_interpreter_always_uses_same_library_instance(self):
+        interpreter1 = nifake._library_interpreter.LibraryInterpreter('windows-1251')
+        interpreter2 = nifake._library_interpreter.LibraryInterpreter('windows-1251')
+        assert interpreter1 is not interpreter2
+        assert interpreter1._library is interpreter2._library
+
+    def test_set_runtime_environment_is_called_if_present(self):
+        self.patched_library_singleton_get.stop()
+        self.patched_library_singleton_lib = patch('nifake._library.Library', return_value=self.patched_library)
+        self.patched_library.niFake_SetRuntimeEnvironment.side_effect = self.side_effects_helper.niFake_SetRuntimeEnvironment
+        self.patched_library_singleton_lib.start()
+        self.get_initialized_library_interpreter()
+        self.patched_library.niFake_SetRuntimeEnvironment.assert_called_once()
+        self.patched_library_singleton_lib.stop()
+        self.patched_library_singleton_get.start()
+
+    def test_set_runtime_environment_not_present_in_driver_runtime(self):
+
+        class TypesLibrary:
+            item = ""
+
+        self.patched_library_singleton_get.stop()
+        self.patched_library = self.PatchedLibrary(TypesLibrary)
+        self.patched_library_singleton_lib = patch('nifake._library.Library', return_value=self.patched_library)
+        self.patched_library_singleton_lib.start()
+        try:
+            # Call _get_library_function directly so that exception is not caught in _library_singleton get
+            self.get_initialized_library_interpreter()._library._get_library_function('niFake_SetRuntimeEnvironment')
+            assert False
+        except nifake.errors.DriverTooOldError:
+            pass
+        nifake._library_singleton._instance = None
+        self.get_initialized_library_interpreter()
+
+        self.patched_library_singleton_lib.stop()
+        self.patched_library_singleton_get.start()
+
+    def test_set_runtime_environment_not_present_in_library(self):
+        self.patched_library_singleton_get.stop()
+        delattr(self.patched_library, 'niFake_SetRuntimeEnvironment')
+        self.patched_library_singleton_lib = patch('nifake._library.Library', return_value=self.patched_library)
+        self.patched_library_singleton_lib.start()
+        nifake._library_singleton._instance = None
+        interpreter = self.get_initialized_library_interpreter()
+        try:
+            interpreter._library.niFake_SetRuntimeEnvironment('', '', '', '')
+            assert False
+        except nifake.errors.DriverTooOldError:
+            pass
```

Comment:
What's going on here?
Are you tring to confirm that DriverTooOldError is raised, as a way to validate that we're properly simulating the abasence of the function from the runtime?

If so, I recommend using `with pytest.raises(nifake.errors.DriverTooOldError):` instead of try, except.
See https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions